### PR TITLE
Move Android test as jvm test: TabDataRepository, and fix flaky test

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/1202552961248957/1208839731155341/f 

### Description

Fixes flaky test due time reference.
Moves it as jvm.

### Steps to test this PR

_Feature 1_
- [ ] CI is green
- [ ]

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|

### CODE REVIEW

**Comment by cmonfortep** [comment]
* **#5316** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5316?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: app/src/test/java/com/duckduckgo/tabs/model/TabDataRepositoryTest.kt [hunk 1]
```diff
@@ -468,7 +485,7 @@ class TabDataRepositoryTest {
     @Test
     fun getActiveTabCountReturnsCorrectCountWhenTabsYoungerThanSpecifiedDay() = runTest {
```
Comment (cmonfortep) [inline app/src/test/java/com/duckduckgo/tabs/model/TabDataRepositoryTest.kt:486]: @0nko after fixing the flaky test, this one is now failing. Can you evaluate the expected result?

File: app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt [hunk 2]
```diff
@@ -199,7 +199,7 @@ class TabDataRepository @Inject constructor(
     }
 
     override fun countTabsAccessedWithinRange(accessOlderThan: Long, accessNotMoreThan: Long?): Int {
-        val now = LocalDateTime.now(ZoneOffset.UTC)
+        val now = timeProvider.localDateTimeNow()
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt:202]: @0nko please review this change, moving to use an injected TimeProvider.

**Review by 0nko — COMMENTED** [review]
Thank you @cmonfortep! I left a couple of comments that in the code to fix the failing test and one more thing.

File: app/src/test/java/com/duckduckgo/tabs/model/TabDataRepositoryTest.kt [hunk 3]
```diff
@@ -497,7 +514,7 @@ class TabDataRepositoryTest {
     @Test
     fun getInactiveTabCountReturnsCorrectCountWhenAllTabsOlderThanSpecifiedDay() = runTest {
         // Arrange: Add some tabs with different last access times
-        val now = LocalDateTime.now(ZoneOffset.UTC)
+        val now = now()
         val tab1 = TabEntity(tabId = "tab1", lastAccessTime = now.minusDays(8))
         val tab2 = TabEntity(tabId = "tab2", lastAccessTime = now.minusDays(10))
         val tab3 = TabEntity(tabId = "tab3", lastAccessTime = now.minusDays(9))
```
Comment (0nko) [inline app/src/test/java/com/duckduckgo/tabs/model/TabDataRepositoryTest.kt:520]: The fix
```suggestion
        val tab3 = TabEntity(tabId = "tab3", lastAccessTime = now.minusDays(9).minusSeconds(1))
```

File: app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt (hunk 2 above)
Comment (cmonfortep) [reply app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt:202]: ok, removing default param as it can't be injected if we do that.

**Review by 0nko — APPROVED** [review]
Looks good, thank you for the fix @cmonfortep! :shipit:

**Pull request merged**